vowpalwabbit/label_parser.h

```diff
@@ -16,4 +16,5 @@ struct label_parser
   float (*get_weight)(void*);
   void (*copy_label)(void*,void*); // copy_label(dst,src) performs a DEEP copy of src into dst (dst is allocated correctly).  if this function is nullptr, then we assume that a memcpy of size label_size is sufficient, so you need only specify this function if your label constains, for instance, pointers (otherwise you'll get double-free errors)
   size_t label_size;
+  int (*get_int_label)(void*);
```

Comment:
Why is this needed?

Reply:
In order to apply class weight, we need to know example's label. This is a way to get the label regardless of example type (simple_label and multiclass store label differently).

Reply:
The traditional method would be get the datatype in the arguments, but we actually have some type information here:
https://github.com/JohnLangford/vowpal_wabbit/blob/master/vowpalwabbit/learner.h#L113 
now.  

You should be able to create the base learner and then observe it's type.  It's also important that the reduction set it's type appropriately with the init() call.

Reply:
ah, do you mean add a new templater parameter to predict_or_learn that specifies how to get the label? so far I've done it at runtime. I can update to templated one.

Reply:
I'd template myself---it should never vary once the reduction stack is instantiated.